core-services/prow/02_config/_config.yaml

```diff
@@ -256,8 +256,6 @@ branch-protection:
           required_status_checks:
             contexts:
             - ci.ext.devshift.net PR build
-        odo:
```

Comment:
This change doesn't enable anything. Please check: https://docs.ci.openshift.org/docs/architecture/branch-protection/

Reply:
Keeping protect: false does not work as it was working earlier - I have no clue to explain it. Recently we came across an issue where prow merges the pr ignoring the failure status of a required job.
Thanks for sharing the doc. I have had gone through the doc before doing the current changes. As per the doc 

Required jobs | protect: stanza | What branchprotector does | Requires admin
-- | -- | -- | --
yes | absent | Enforces BP for required jobs | yes

I am removing the odo field from here so that prow starts managing the default BP settings for odo repo.
Please feel free to comment on my understanding

Reply:
@droslean Could you clarify and correct me If I am wrong in https://github.com/openshift/release/pull/16380#discussion_r586302974

Reply:
There is no required job configured now. 
You should be able to do something like
```
          required_status_checks:
            contexts:
            - my-required-job
```

Reply:
@mohammedzee1000 @kadel ptal

Reply:
The title of this PR is not accurate and it might lead to some confusion. 

We used to manage branch protection in GitHub settings because we had external CI jobs (travis) that we needed to be required. This we before it was possible to set external jobs as required here. 

We don't use Travis anymore and we want to get back to managing all jobs in Prow.

This is what this PR is supposed to do. 
@droslean Is this the right configuration that will allow us to simply set required or optional jobs using `optional`  field in the job configuration?

Reply:
@kadel The `optional` field in the job configuration is not related to the branch protection settings. If you want to manage only that, you should leave the `protect: false` here as it is and just simply mark as `optional` any jobs you need.

Reply:
Just to clarify. If you configure the branch protection to have required jobs, you will use Github's branch protection features to protect the branch from merging if a required job fails. However, our CI auto merges the PRs without any human integration, and just setting the `optional` field in the job configuration will serve your purpose.

Reply:
Sorry for misleading title of the pr. I just changed it to clarify the pr agenda

Reply:
@droslean The current configuration  (`protect: false`) makes all our jobs optional, and if we try to configure it in GitHub's branch protection, it disables it after some time. Which based on https://docs.ci.openshift.org/docs/architecture/branch-protection/ is expected. But that is not what we want.

> However, our CI auto merges the PRs without any human integration, and just setting the optional field in the job configuration will serve your purpose.

This is what we want. 

So what should be the correct `branch-protection:` setting here?

Reply:
My understanding from the above thread 

- The optional field in the job configuration has nothing to do with branch protection 
- Prow does not manage branch label protection 
- Branch label protection is managed by GitHub account and prow listen to that

So the solution to avoid merge of a failed pr reported in any one job to is
Enable branch label protection in project GitHub account and make that job as required in the GitHub BP page plus in the job config make the same job required (Default).
@droslean Please confirm.

Reply:
@kadel The current configuration doesn't make your jobs optional. @prietyc123 The solution is not even related to branch protection settings. You will have to mark `optional: true` the jobs that you need to be optional, and that's it.